### PR TITLE
Feature/SALEOR-3109 Use pagination on attribute values

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it adds pagination for attribute values.

**PR intended to be tested with API branch:**  SALEOR-3071-paginate-attributes
https://github.com/mirumee/saleor/pull/7341

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [x] Attributes `[data-test-id]` are added for new elements
4. [x] Changes are mentioned in the changelog
5. [x] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://saleor-3071-paginate-attributes.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3109 Use pagination on attribute values](https://app.clickup.com/t/2549495/SALEOR-3109)

**Review by pwgryglak — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

File: src/hooks/useLocalPaginator.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  hasPreviousPage: boolean;
+  startCursor: string;
+}
+
+export interface PaginationState {
+  after?: string;
+  before?: string;
+  first?: number;
+  last?: number;
+}
```
Comment (dominik-zeglen) [inline src/hooks/useLocalPaginator.ts:15]: I believe I didn't get to review this piece of code but `PageInfo` and `PagiationState` types are defined somewhere in the app and I don't think they should be redefined here

File: src/hooks/useLocalPaginator.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      });
+    } else if (paginationState.before) {
+      setState({
+        before: paginationState.before,
+        last: paginateBy
+      });
+    } else {
+      setState({
+        first: paginateBy
+      });
```
Comment (dominik-zeglen) [inline src/hooks/useLocalPaginator.ts:38]: I would also like to integrate it with `usePaginator` to keep this logic in one place even if callbacks are different (`setState` vs `navigate`)

**Review by dominik-zeglen — APPROVED** [review]

File: src/hooks/useLocalPaginator.ts (hunk 1 above)
Comment (orzechdev) [reply src/hooks/useLocalPaginator.ts:15]: will be changed in other pr

**Pull request merged**